### PR TITLE
Fixing logging and unnecessary parameter check in prepare-key script …

### PR DESCRIPTION
**Pull request by ibaldin**
…- AMI/AKI/ARI parameters aren't needed and aren't supplied when modify is called (as opposed to creating a new instance)

### CODE REVIEW

**Comment by ibaldin** [comment]
I fixed a bunch of indentations and excessive logging in the ANT handler. The main change is really removing the if: statement in prepare-key script (and indenting everything back, since this is Python). The check is useless and makes the script fail in modify case (rather than create new instance) because ANT can't supply the AKI/AMI/ARI.

**Comment by ibaldin** [comment]
Is it possible to approve this and deploy on some rack before cutting the release?

File: handlers/ec2/resources/scripts/nova-essex-prepare-key [hunk 1]
```diff
@@ -56,41 +56,38 @@ try:
     #DO THE WORK                                                                                                                                                                                       
     LOG.debug("nova-essex-prepare-key: " + str(sys.argv))
 
-    if str(os.environ['AMI_NAME']) and str(os.environ['AKI_NAME']) and str(os.environ['ARI_NAME']):
```
Comment (ibaldin) [inline handlers/ec2/resources/scripts/nova-essex-prepare-key]: This is the main change

File: handlers/ec2/resources/handlers/ec2/handler.xml [hunk 2]
```diff
@@ -653,9 +647,6 @@
 								<env key="EUCA_KEY_DIR" value="${ec2.keys}" />
 								<env key="EC2_SSH_KEY" value="${ec2.ssh.key}" />
 								<env key="EC2_SSH_TIMEOUT" value="${ec2.ssh.timeout}" />
-								<env key="AMI_NAME" value="${ec2.ami.name}" />
```
Comment (ibaldin) [inline handlers/ec2/resources/handlers/ec2/handler.xml]: Not needed

File: handlers/ec2/resources/handlers/ec2/handler.xml [hunk 3]
```diff
@@ -591,38 +588,35 @@
...
+						<env key="EC2_LOG_DIR" value="${ec2.log.dir}" />
+						<env key="EC2_LOG_FILE" value="${ec2.log.file}" />
+						<env key="EC2_LOG_LEVEL" value="${ec2.log.level}" />
+						<env key="EC2_DIR" value="${ec2.dir}" />
+						<env key="EUCA_KEY_DIR" value="${ec2.keys}" />
+						<env key="EC2_SSH_KEY" value="${ec2.ssh.key}" />
+						<arg value="${unit.ec2.instance}" />
+					</exec>
+					<echo message="Exit code for get-userdata is ${icode}" />
+					<!-- <echo message="Old userdata : ${userdata.old}" /> -->
```
Comment (ibaldin) [inline handlers/ec2/resources/handlers/ec2/handler.xml]: Excessive logging.

**Comment by hinchliff** [comment]
This looks like it reverts changes from #138

We will need to make sure we don't reintroduce the problems addressed in that pull request.

File: handlers/ec2/resources/scripts/nova-essex-prepare-key (hunk 1 above)
Comment (mcevik0) [reply handlers/ec2/resources/scripts/nova-essex-prepare-key]: I had inserted this statement to by-pass SSH key insertion for Windows images. It was my presumption that Windows images would have only AMI whereas Linux images would have AMI,AKI,ARI. I will find a better way for Windows images for inserting SSH keys. (Probably install an SSH server to the image.)

**Comment by mcevik0** [comment]
Since the main change is removing the if statement, these changes probably will not introduce problems with Unix-based images. That statement was inserted just to by-pass key insertion for the Windows VMs. I will prepare the environment for testing.

**Comment by mcevik0** [comment]
Deployed to GWU for testing. It works as expected. 
https://github.com/RENCI-NRIG/orca5/issues/169#issuecomment-351799136

**Pull request merged**